test/sha2.test.mo

```diff
@@ -281,3 +281,33 @@ do {
   assert(digest256.sum() == h256);
   assert(digest512.sum() == h512);
 };
+
+// test share() and unshare()
+
+do {
+  let data = Blob.fromArrayMut(Array.init<Nat8>(64, 0));
+  var digest256 = Sha256.Digest(#sha256);
+  var digest512 = Sha512.Digest(#sha512);
+
+  var digest256_state = digest256.share();
+  var digest512_state = digest512.share();
+
+  for (i in range(1,10000)) {
```

Comment:
Is there any particular reason to share/unshare data 10000 times in the test? Can we ensure that share/unshare work correctly by performing it once or twice?